org.eclipse.jdt.debug.ui/ui/org/eclipse/jdt/internal/debug/ui/ObjectComparision.java

```diff
@@ -0,0 +1,654 @@
+	 * @param selectedObject1
+	 *            First selected object from variable view
+	 * @param selectedObject2
+	 *            Second selected object from variable view
+	 * @throws DebugException
+	 * @return returns a Map of comparison result details
+	 */
+	private Map<String, Object> compareConcurrentSet(IJavaObject val1, IJavaObject val2, IJavaVariable selectedObject1, IJavaVariable selectedObject2) throws DebugException {
+		IJavaObject root1 = (IJavaObject) val1.getVariables()[1].getValue();
+		IJavaObject root2 = (IJavaObject) val2.getVariables()[1].getValue();
+		List<Object> values1 = new ArrayList<>();
+		List<Object> values2 = new ArrayList<>();
+		Map<String, Object> result = new HashMap<>();
+		if (root1 instanceof JDIArrayValue jdiArray1 && root2 instanceof JDIArrayValue jdiArray2) {
+			for (IJavaValue jdiValue : jdiArray1.getValues()) {
+				if (!(jdiValue instanceof JDINullValue)) {
+					values1.add(objectExtraction(jdiValue));
+				}
+			}
+			for (IJavaValue jdiValue : jdiArray2.getValues()) {
+				if (!(jdiValue instanceof JDINullValue)) {
+					values2.add(objectExtraction(jdiValue));
+				}
+			}
+			result.put(SELECTED_OBJECT_1, selectedObject1.getName());
+			result.put(SELECTED_OBJECT_2, selectedObject2.getName());
+			result.put(VALUESET_1, values1);
+			result.put(VALUESET_2, values2);
+			return result;
+		}
+		result.put(IMMEDIATE_RESULT_KEY, false);
+		String failed = "Selected type is not IJavaValue type"; //$NON-NLS-1$
+		result.put(RESULT_FAILED, failed);
+		return result;
+
+	}
+
+	/**
+	 * Compare Normal objects
+	 * @param selectedObject1
+	 *            First selected object from variable view
+	 * @param selectedObject2
+	 *            Second selected object from variable view
+	 * @throws DebugException
+	 * @return returns a Map of comparison result details
+	 */
+	@SuppressWarnings("nls")
+	public Map<String, Object> compareObjects(IJavaVariable selectedObject1, IJavaVariable selectedObject2) throws DebugException {
+		Map<String, Object> result = new HashMap<>();
+		result.put(SELECTED_OBJECT_1, selectedObject1.getName());
+		result.put(SELECTED_OBJECT_2, selectedObject2.getName());
+		boolean res = false;
+		if (selectedObject1.getValue() instanceof IJavaValue javaVal1 && selectedObject2.getValue() instanceof IJavaValue javaVal2) {
+			String v1 = objectExtraction(javaVal1);
+			String v2 = objectExtraction(javaVal2);
+			if (v1 == null || v2 == null) {
+				res = javaVal1.toString().equals(javaVal2.toString());
+				result.put(IMMEDIATE_RESULT_KEY, res);
+				return result;
+			}
+			res = v1.equals(v2);
+			if (!res) {
+				result.put(IMMEDIATE_RESULT_KEY, res);
+				String failureReason = "Selected objects are different ";
+				result.put(RESULT_FAILED, failureReason);
+				return result;
+			}
+			result.put(IMMEDIATE_RESULT_KEY, res);
+			return result;
+		}
+		result.put(IMMEDIATE_RESULT_KEY, res);
+		String failed = "Selected type is not IJavaValue type";
```

Comment:
Who would see this message and when? A user could not understand the meaning.

Reply:
removed it